node/src/tasks/bsp_volunteer_mock.rs

```diff
@@ -0,0 +1,118 @@
+use std::str::FromStr;
+
+use log::{debug, error, info};
+use sp_core::H256;
+use sp_runtime::BoundedVec;
+use storage_hub_infra::{actor::ActorHandle, event_bus::EventHandler};
+
+use crate::services::{
+    blockchain::{
+        commands::BlockchainServiceInterface, events::NewStorageRequest, handler::BlockchainService,
+    },
+    StorageHubHandler,
+};
+
+const LOG_TARGET: &str = "bsp-volunteer-mock-task";
+
+#[derive(Clone)]
+pub struct BspVolunteerMockTask {
+    storage_hub_handler: StorageHubHandler,
+}
+
+impl BspVolunteerMockTask {
+    pub fn new(storage_hub_handler: StorageHubHandler) -> Self {
+        Self {
+            storage_hub_handler: storage_hub_handler,
+        }
+    }
+}
+
+impl EventHandler<NewStorageRequest> for BspVolunteerMockTask {
+    async fn handle_event(&self, event: NewStorageRequest) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Initiating BSP volunteer mock for location: {:?}, fingerprint: {:?}",
+            event.location,
+            event.fingerprint
+        );
+
+        // TODO: Here we would send the actual multiaddresses of this BSP.
+        let multiaddresses = BoundedVec::default();
+
+        // Build extrinsic.
+        let call =
+            storage_hub_runtime::RuntimeCall::FileSystem(pallet_file_system::Call::bsp_volunteer {
+                location: event.location,
+                fingerprint: event.fingerprint,
+                multiaddresses,
+            });
+
+        let (mut tx_watcher, tx_hash) = self
+            .storage_hub_handler
+            .blockchain
+            .send_extrinsic(call)
+            .await;
+
+        // Wait for the transaction to be included in a block.
+        let mut block_hash = None;
+        // TODO: Consider adding a timeout.
+        while let Some(tx_result) = tx_watcher.recv().await {
+            // Checking if there is an update with an error in our transaction.
+            if tx_result.starts_with("Error") {
```

Comment:
Should we change the receiver channel to have a `Result` instead and automatically return `Err` variant if “Error” is in the transaction result since it will be done for all tasks and probably be repeated?

Reply:
Yes, good call, will do

Reply:
Good call, will do